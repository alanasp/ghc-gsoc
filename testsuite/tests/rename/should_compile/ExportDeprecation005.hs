{-# OPTIONS_GHC -fwarn-warnings-deprecations #-}

-- Test deprecation of variable and data constructor exports

module ShouldCompile where

import ExportDeprecation005_B(func_C) --deprecated
import ExportDeprecation005_C(func_C) --non-deprecated
import ExportDeprecation005_A(func_C) --deprecated

main = func_C --no warning because non-deprecated import declaration exists
