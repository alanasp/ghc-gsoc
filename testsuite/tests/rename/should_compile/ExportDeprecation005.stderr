ExportDeprecation005.hs:7:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "ExportDeprecation005_C shall be imported directly"

ExportDeprecation005.hs:9:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "ExportDeprecation005_C shall be imported directly"
